### ISSUE TITLE
🚧Setup CICD for project

### ISSUE THREAD

**Issue by CalvinWilkinson** [KinsonDigital/Plazma#1]
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Update the _**CICD**_ dotnet tool to the latest version.   This will require various _**dotnet**_ commands to be able to set up the _**kinsondigital.cicd**_ dotnet tool that is globally in NuGet.

💡What is a **_dotnet_** tool?
A dotnet tool is nothing but a console application that anybody can create that you can easily install and use in another dotnet application.  As long as the _**dotnet**_ tool is publicly available in nuget.org, it can be automatically pulled into your computer or installed globally on your machine to use!!  It is free and make it easy to distribute tools for anything from simple one off project use cases to complicated CICD automation.

### **Installing CICD**
1. Create a manifest file.
    > **Note** This is nothing but a configuration file. This config file is just a list of all the **_dotnet_** tools that are setup in a project for the _**dotnet**_ ecosystem to use.
    1. If you do not have the **_dotnet_** CLI installed on your machine, we first need to get that installed.  First check to see if you have the CLI installed by opening the windows terminal in ANY location.  Execute the command `dotnet --version`.  If you see a version number then you have it installed.
    2. Next, create a tool manifest file.  This will create an barebones file in root of the project so you can installed _**dotnet**_ tools.  
        To do this, run the command `dotnet new tool-manifest` the root of the repository.  Once it is complete, you will see a _**.config**_ folder created in the root of the repo for you and inside you should see a **_dotnet-tools.json_** file inside it.
2. Now it is time to add a dotnet tool to the project.  
    1. To create a dotnet tool, use the command `dotnet tool install kinsondigital.cicd --version 1.0.0-preview.20`.
        This should update the _**dotnet-tools.json**_ manifest file with additional information.  That info is the name of the tool as well as the version.  This gives the user the ability to run the `dotnet tool restore` command in the same location as the _**.config**_ folder at any time and it will automatically install the tool for that project so it is ready to use.

### **CICD Configuration**

This configuration is what **_CICD_** uses to setup some of its behavior when it runs.  Follow the steps below to set this up.

1. Create a folder in the root of the repository with the name _**.nuke**_.
2. Inside of the _**.nuke**_ folder, create a new file with the name _**parameters.json**_.
3. Add the JSON content below to the file from step 2.
```json
{
    "$schema": "./build.schema.json",
    "Solution": "Partithyst.sln",
    "RepoOwner": "KinsonDigital",
    "RepoName": "Partithyst",
    "ProjectName": "Partithyst",
    "PreProcessReadMe": true
}
```
4. If everything is setup correctly, you should be able to run the `dotnet cicd --help` and see results similar to the image below.
![Image](https://user-images.githubusercontent.com/85414302/228282862-aa715471-5bdb-4193-a753-c92be325582d.png)
5. Auto generate the workflow files using a CICD command.
    1. This can be done by executing the `dotnet cicd GenerateWorkflows --workflow-template-output "<FULL FILE PATH>"`
        > **Note** The path to generate will be a directory path and it MUST BE A FULLY QUALIFIED PATH.  Use the full path to the **_.github/workflows_** directory.  If this does not exist, create it first then run the command.
7. To create the badges, just copy prev and prod PR status badges from Velaptor to this project.
> **Note** Once you copy the badges over, just change the name of the projects in the URLs.  Make sure that the names of the workflow _***.yml**_ files in the URL are correct.  These are the workflows you generated in step 5.

### Acceptance Criteria

- [x] CICD setup for the project
- [x] Badges for the prev and prod release PR status check setup in the README file.
- [x] Workflows generated using the CICD dotnet tool.

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct project _(if applicable)_.
- [X] Issue linked to the correct milestone _(if applicable)_.
- [x] Draft pull request created and linked to this issue _(only required with code changes)_.

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| CICD Changes        | `♻️cicd` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.
